### PR TITLE
Set up cs-ranking for release on PyPi

### LINKED ISSUE #69: Prepare package for PyPI/Anaconda
Rationale
---------
For users it is much more convenient to be able to install the package from PyPI using a simple
```shell
pip install cs-ranking
```
or
```shell
conda install -c conda-forge cs-ranking
```
rather than checking out the repository.

What needs to be done
---------------------
- Check the official [packaging guidelines](https://packaging.python.org/tutorials/packaging-projects/).
- Create a [recipe](https://github.com/conda-forge/staged-recipes) for conda-forge.

### LINKED ISSUE #70: Implement versioneer support for automatic version string updates
Rationale
---------
Updating the version string when doing a new release is error-prone. [Versioneer](https://github.com/warner/python-versioneer) automatically determines the version string by querying git.
This allows us to simply tag a commit using a descriptive name like `v1.1` or `1.1` and it will be applied automatically.

### PR DESCRIPTION
**Pull request by kiudee**
## Description
The following changes are implemented:

* Usage of [bump2version](https://github.com/c4urself/bump2version) to semi-automated changing of version strings
* Add token for PyPi to .travis.yml
* Create HISTORY.rst file to track changes

## Motivation and Context
To prepare the release of the next minor version 1.1.0, we should get everything set up.

## How Has This Been Tested?
Executed `bump2version patch` to test if it correctly updates all the version strings.

## Does this close/impact existing issues?
Resolves #69, closes #70

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -28,5 +28,16 @@ script:
   - travis-sphinx build -n --source=docs
 
 after_success:
-  - coveralls
-  - travis-sphinx deploy
+- coveralls
+- travis-sphinx deploy
```
[the remark this answers is not in the document]
Comment (kiudee) [reply .travis.yml]: Indeed, that was caused by the travis-encrypt tool.

File: .travis.yml [hunk 2]
```diff
@@ -28,5 +28,16 @@ script:
...
+
+deploy:
+  provider: pypi
+  distributions: sdist bdist_wheel
+  user: __token__
+  password:
+    secure: INa6/h17ejVh3vxAGc4V7uLY30tJON2n/PWV3y0x0DnAR+pH5LiR3ZIE8/CNu1NpH+R7I1u2m0PHLYh5H8XpuewHaToer4p41jIiGF6WXnUg6PQN7zuBk9tqgmJElr+aYp/oK/B6VHTfFwEk2QlXmcqVofpKQNWrzm6+EMWXJjSs3Z9XyGIJvzJ2ihaHvr4URlNfjX98Ij0uIzzEBzN7RSJISQQo672F0b+JvgR3nQYfa5Sp4ijBnEuTtU2BYiXcojaT283O49DL0LGQHApIu0Blf79FXA2QJET65ujIQDxKBZ+WhPEHupwIiBsJ1JLl5C0wBj0UAXyogypXhm9tXfOwB12MLsbT8pbH7YMY4vS8lKtXltWTw66JbruNxECCcvHMw6SnCX0D1H40LKCvQLLJ4L2IOkVcpLI702Tlf13OF4kic+OGfSBv0nRDTLzyvMU5NACoMLnBipbpN4qXKeJdoHWOKECiRYFqZLacxPSGXx/RteiEpi25Ghgk2x53Y6pNCpzW118qW/Ij2LvPffB9PT/BPe0MIjveDLJ3PI93xevDljyZHE8v7JGOPo4gX3YA4hYwMB6gZtO+bdIemGQIP5ymrIpKqSDw/CpFLUCUVzL8sOu5AE3pbDFUGdn5njSex7Leafs+px7IH6dLE2MQqCU1o9Qr4V4FTuxta60=
+  true:
```
[the remark this answers is not in the document]
Comment (kiudee) [reply .travis.yml]: Yes, `on:` is correct.

File: setup.py [hunk 3]
```diff
@@ -2,25 +2,33 @@
 
 from setuptools import setup, find_packages
 
-DISTNAME = 'csrank'
-DESCRIPTION = 'Context-sensitive ranking'
-MAINTAINER = 'Karlson Pfannschmidt'
-MAINTAINER_EMAIL = 'kiudee@mail.upb.de'
-VERSION = "1.0"
+with open("README.rst") as readme_file:
```
[the remark this answers is not in the document]
Comment (kiudee) [reply setup.py]: Yes, I decided to have one pull request for all the changes necessary to get proper version numbers and deployment running (bump2version, history file, travis deploy).
We could have done it more fine grained, I agree.

**Comment by timokau** [comment]
Looks good to me now. I still think some documentation would be good, especially if we want to have a broader contributor base in the future. But that could also be done as a follow-up.

**Pull request merged**